gradio/routes.py

```diff
@@ -304,6 +304,8 @@ async def run_predict(
                 output["data"] = output["data"][0]
             return output
 
+        @app.post("/run/{api_name}", dependencies=[Depends(login_check)])
```

Comment:
Consider adding a comment here for future readers of the codebase explaining why we have a second route

Reply:
Done